SGoopas/Assets/Scripts/Game/2D/2D Player States/AirDashLeft2D.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayerStates
+{
+    public class AirDashLeft2D : Base2DState
+    {
+        private float afterImageTime = 0.1f;
+        private float timeEllapsed;
+        private GameObject afterimage;
+        private Vector3 direction;
+        private IEnumerator coroutine;
+        private Rigidbody2D rigidbody;
+        private float origGrav;
+        public AirDashLeft2D(BasePlayerState previousState) : base(previousState)
+        {
+            direction = new Vector3();
+            PseudoConstructor();
+        }
+        public AirDashLeft2D(GameObject player, MasterPlayerStateMachine playerStateMachine, Transform groundCheck) : base(player, playerStateMachine, groundCheck)
+        {
+            direction = new Vector3();
```

Comment:
What happens if you dash with a (0, 0, 0) direction vector?

Reply:
You just sorta hover in place for the dash time. .normalized returns a (0,0,0) vector for (0,0,0) vectors, so your velocity is 0 during the dash. I could set the default constructor to a forward vector though if we'd prefer